### ISSUE TITLE
Training works only with gray scale images

### ISSUE THREAD

**Issue by kalai2033** [junyanz/pytorch-CycleGAN-and-pix2pix#825]
Hi,
   I have used pix2pix model to train my model using custom dataset with pairs. A is the real image while B is the black & white version of A. When i trained the model with 3 channel ie RGB image(--n_channels 3), the model does not learn anything but just outputs a black image. But when i use gray scale images(--n_channels 1), it learns perfectly. Btw, i use resnet9 for G and basic for D. Can you please explain the behaviour?

**Comment by junyanz** [comment]
Did you set the correct ``-input_nc` and `--output_nc`for your first experiment?

**Comment by kalai2033** [comment]
Yeah i set the params correctly. These are my sample images....

![Image A)](https://user-images.githubusercontent.com/49688647/68300748-af0e3c80-009e-11ea-819a-3e83460a77b8.JPG)


![Image B](https://user-images.githubusercontent.com/49688647/68300697-9736b880-009e-11ea-8d12-c72b582e4d3b.jpg)

**Comment by junyanz** [comment]
if your output image is a binary image, it may cause trouble for the default GAN loss. It is easy for the discriminator to distinguish ground truth binary data and generated results. You may want to try other GAN losses (e.g., `--gan_mode lsgan`).

**Comment by kalai2033** [comment]
@junyanz Thanks.. I will try it out
